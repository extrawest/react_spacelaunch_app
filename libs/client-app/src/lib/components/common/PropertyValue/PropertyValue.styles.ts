import type { SxProps } from '@mui/material';

const property: SxProps = {
  fontWeight: 700,
};

export const styles = {
  property,
};
