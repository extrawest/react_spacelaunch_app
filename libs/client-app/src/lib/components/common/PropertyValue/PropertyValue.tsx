import type { FC } from 'react';

import { Box, Typography } from '@mui/material';

import { styles } from './PropertyValue.styles';
import type { PropertyValueProps } from './PropertyValue.types';

export const PropertyValue: FC<PropertyValueProps> = ({
  property,
  children,
}) => {
  return (
    <Typography variant="body1">
      <Box
        component="span"
        sx={styles.property}
      >
        {property}:
      </Box>
      &nbsp;
      {children}
    </Typography>
  );
};
