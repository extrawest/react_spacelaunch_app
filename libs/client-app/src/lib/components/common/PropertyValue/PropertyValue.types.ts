export interface PropertyValueProps {
  property: string;
  children: string;
}
