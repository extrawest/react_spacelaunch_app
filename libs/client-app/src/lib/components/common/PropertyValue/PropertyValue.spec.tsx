import { render } from '@testing-library/react';

import { PropertyValue } from './PropertyValue';

describe('[component]: PropertyValue', () => {
  const property = 'property';
  const value = 'value';

  test('Should match default style', () => {
    const { baseElement } = render(
      <PropertyValue property={property}>{value}</PropertyValue>
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Property and value should be correct', () => {
    const { getByText } = render(
      <PropertyValue property={property}>{value}</PropertyValue>
    );
    const formattedProperty = `${property}:`;

    expect(getByText(formattedProperty)).toHaveTextContent(formattedProperty);
    expect(getByText(value)).toHaveTextContent(value);
  });
});
