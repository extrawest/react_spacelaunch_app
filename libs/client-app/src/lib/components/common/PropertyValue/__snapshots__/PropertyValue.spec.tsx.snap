// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]: PropertyValue Should match default style 1`] = `
<body>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <span
        class="MuiBox-root css-1vg6q84"
      >
        property
        :
      </span>
      Â 
      value
    </p>
  </div>
</body>
`;
